### DISCUSSION TITLE
Custom Directory for Builds

### DISCUSSION THREAD

**Discussion by ChrisJBurns** [concourse/concourse#8763] [General]
Am wondering if it's currently possible to define where the build directory get's configured. In our Maven image that we use for maven tasks of building Java projects, we assign a user called `builduser` and a directory `/builduser` so that we can control the permissions of the user in what it can or can't do. However, Concourse create the builds folders in `/tmp/build/$random-hash/...`. Is there an option where we can tell Concourse where to run the builds? So in our case, we want it to run the builds in the `/builduser` directory. So something like `/builduser/build/$randomhash`?

**Comment by taylorsilva** [comment]
Sadly there is no way to specify where the build directory should be located. Whenever I run into this pain I usually end up with this bit at the top of my task scripts.

```sh
BUILD_DIR="$(pwd)"
```

and then I'll prefix `$BUILD_DIR` to anything that needs an absolute instead of relative path.

**Reply by ChrisJBurns** [reply]
I found [this issue](https://github.com/concourse/concourse/issues/403) after I asked this discussion item which makes me feel that it isn't going to be addressed anytime soon. The reason I wanted to be able to specify where the build directory get's put is because I wanted to control the permissions of that directory. But as described on the issue I've linked, nonroot build directories aren't possible yet.

**Comment by jriddy** [comment]
What does this do then? `task-config.run.dir` https://concourse-ci.org/tasks.html#schema.command

**Reply by ChrisJBurns** [reply]
That's just the working directory to which the task will start within. It doesn't control the path to which inputs are created
